server/src/metadata.js

```diff
@@ -231,8 +231,9 @@ class Metadata {
       // Redirect the 'users' table to the one in the internal db
       this._collections.set('users', new Collection({ id: 'users', table: 'users' },
                                                     this._internal_db, this._conn));
-    }).catch(() => {
+    }).catch((e) => {
       this.close();
+      throw e;
```

Comment:
I think this should just chain onto `this._ready_promise` rather than replacing it.

Like:

```
    [...]
    }).then(() => {
      [...]
    })

    this._ready_promise.catch(() => {
      this.close();
    });
```